### DISCUSSION TITLE
generating html with inline images with mochawesome

### DISCUSSION THREAD

**Discussion by ToAM90** [cypress-io/cypress#29096] [Questions and Help]
Hi, I am currently trying to create and email a html report with screenshots via mochawesome. In order for it to be emailed, I have been trying to turn the png images into base64 to inline it into the html file. However, I am not able to do so successfully... I have tired different methods to encode the png to base64, such as FileReader, cy.readFile, fs.readFileSync, image-to-base64 npm package, etc with not much success. Wondering if theres a specific way to be able to do this for Cypress and if anyone can help. 

```
Cypress.on('test:after:run', (test, runnable) => {
  if(test.state === 'failed' && test.currentRetry === test.retries){
    const title = test.title.replace('.', "").replace('#', '%23');
    const parentTitle = runnable.parent.title.replace(".", "");
    const filePath = (test.retries == 0) ? 
      `${Cypress.config('screenshotsFolder')}/${Cypress.spec.name}/${parentTitle} -- ${title} (failed).png`
      : `${Cypress.config('screenshotsFolder')}/${Cypress.spec.name}/${parentTitle} -- ${title} (failed) (attempt 2).png`;
    const fs = require("fs")
    const pngAsBase64 = fs.readFileSync(filePath, "base64")
    addContext({ test }, "data:image/png;base64," + pngAsBase64);
  };
})
```

**Comment by portyu9** [comment]
Hi @ToAM90 

You do not need to manually read the screenshot files and convert them to base64.

If the goal is a **single self-contained Mochawesome HTML file**, `cypress-mochawesome-reporter` supports this directly:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  reporter: 'cypress-mochawesome-reporter',

  reporterOptions: {
    embeddedScreenshots: true,
    inlineAssets: true,
    saveAllAttempts: false,
  },

  e2e: {
    setupNodeEvents(on) {
      require('cypress-mochawesome-reporter/plugin')(on)
    },
  },
})
```

And in `cypress/support/e2e.js`:

```js
import 'cypress-mochawesome-reporter/register'
```

`embeddedScreenshots` embeds the screenshots as base64, while `inlineAssets` puts the report assets into the HTML file as well.

This is functionality of the third-party `cypress-mochawesome-reporter`, rather than Cypress core, so make sure you select a reporter version compatible with your Cypress/Node versions.

Please mark this answer as accepted if it helped, thank you.
